service-front/module/Application/src/Helpers/LpaFormHelper.php

```diff
@@ -0,0 +1,227 @@
+            }
+
+            $idCheck = $this->compareCpRecords($detailsData, $siriusCheck);
+            $statusCheck = $this->checkStatus($siriusCheck);
+            $channelCheck = $this->checkChannel($siriusCheck);
+
+            if ($idCheck['error'] === true) {
+                $result['status'] = 'no match';
+                $result['message'] = $idCheck['message'];
+                $result['additional_data'] = [
+                    'name' => $idCheck['name'],
+                    'address' => $idCheck['address'],
+                    'name_match' => $idCheck['name_match'],
+                    'address_match' => $idCheck['address_match'],
+                    'error' => $idCheck['error']
+                ];
+            } elseif (! $this->checkLpaNotAdded($form->get('lpa')->getValue(), $detailsData)) {
+                $result['status'] = 'error';
+                $result['message'] = "This LPA has already been added to this ID check.";
+            } elseif ($statusCheck['error'] === true) {
+                $result['status'] = 'error';
+                $result['message'] = $statusCheck['message'];
+            } elseif ($channelCheck['error'] === true) {
+                $result['status'] = 'error';
+                $result['message'] = $channelCheck['message'];
+            } else {
+                $result['status'] = "success";
+                $result['message'] = "";
+            }
+            $result['data'] = [
+                "case_uuid" => $uuid,
+                "lpa_number" => $form->get('lpa')->getValue(),
+                "type_of_lpa" => $this->getLpaTypeFromSiriusResponse($siriusCheck),
+                "donor" => $this->getDonorNameFromSiriusResponse($siriusCheck),
+                "lpa_status" => $statusCheck['status'],
+                "cp_name" => $idCheck['name'],
+                "cp_address" => $idCheck['address']
+            ];
+        }
+
+        return new LpaFormHelperResponseDto(
+            $uuid,
+            $form,
+            /**
+             * @psalm-suppress PossiblyUndefinedArrayOffset
+             */
+            $result['status'],
+            $result['message'],
+            array_key_exists('data', $result) ? $result['data'] : [],
+            array_key_exists('additional_data', $result) ? $result['additional_data'] : [],
+        );
+    }
+
+    private function getLpaTypeFromSiriusResponse(array $siriusCheck): string
+    {
+        try {
+            return $siriusCheck['opg.poas.lpastore']['lpaType'] ?? '';
+        } catch (\Exception $exception) {
+            return $exception->getMessage();
+        }
+    }
+
+    private function getDonorNameFromSiriusResponse(array $siriusCheck): string
+    {
+        try {
+            return $siriusCheck['opg.poas.sirius']['donor']['firstname'] .
+                " " .
+                $siriusCheck['opg.poas.sirius']['donor']['surname'];
+        } catch (\Exception $exception) {
+            return $exception->getMessage();
+        }
```

Comment:
Similarly this can be cleaned up whilst avoiding warnings:
```suggestion
        return implode(' ', [
            $siriusCheck['opg.poas.sirius']['donor']['firstname'] ?? '',
            $siriusCheck['opg.poas.sirius']['donor']['surname'] ?? '',
        ]);
```